File: September-LeetCoding-Challenge/03-Repeated-Substring-Pattern/Repeated-Substring-Pattern.cs
public class Solution {
    public bool RepeatedSubstringPattern(string s) {
        var len = s.Length;
        for (int i = len / 2; i >= 1; i--)
        {
            // Dividable
            if (len % i == 0)
            { 
                int m = len / i;
                var sub = s.Substring(0, i);
                int j;

                //Check sub is repeated in each round, until m rounds.
                for (j = 1; j < m; j++)
                {
                    if (!sub.Equals(s.Substring(j * i, i))) break;
                }

                //If sub is repeated m rounds, then it's a repeated substring
                if (j == m)
                    return true;
            }
        }
        return false;
    }

    public void Test()
    {
        var s = "abcdabcdabcd";
        var r = RepeatedSubstringPattern(s);
        Console.WriteLine(r);
    }
}

File: September-LeetCoding-Challenge/03-Repeated-Substring-Pattern/Repeated-Substring-Pattern.rb
def repeated_substring_pattern(s)
    out = false
    length = s.length / 2 
    0.upto(length) do |n|
        out =  is_repeated_sub?(s[0..n], s)
        return out if out == true
    end
    out
    
end

def is_repeated_sub?(sub, string)
    modulo = string.length % sub.length
    return false if modulo != 0
    multiplicator = string.length / sub.length 
    return false if multiplicator <= 1
    return (sub * multiplicator) == string
end